pkgs/applications/editors/kakoune/plugins/update.py

```diff
@@ -21,30 +21,40 @@
 )
 import pluginupdate
 
-GET_PLUGINS = f"""(with import <localpkgs> {{}};
+GET_PLUGINS = f"""with import <localpkgs> {{ }};
 let
-  inherit (kakouneUtils.override {{}}) buildKakounePluginFrom2Nix;
-  generated = callPackage {ROOT}/generated.nix {{
-    inherit buildKakounePluginFrom2Nix;
+  inherit (kakouneUtils.override {{ }}) buildKakounePluginFrom2Nix;
+  generated = callPackage {ROOT}/generated.nix {{ inherit buildKakounePluginFrom2Nix; }};
+
+  hasChecksum =
+    value:
+    lib.isAttrs value
+    && lib.hasAttrByPath [
+      "src"
+      "outputHash"
+    ] value;
+
+  parse = name: value: {{
+    pname = value.pname;
+    version = value.version;
+    homePage = value.meta.homepage;
+    checksum =
+      if hasChecksum value then
+        {{
+          submodules = value.src.fetchSubmodules or false;
+          sha256 = value.src.outputHash;
+          rev = value.src.rev;
+        }}
+      else
+        null;
   }};
-  hasChecksum = value: lib.isAttrs value && lib.hasAttrByPath ["src" "outputHash"] value;
-  getChecksum = name: value:
-    if hasChecksum value then {{
-      submodules = value.src.fetchSubmodules or false;
-      sha256 = value.src.outputHash;
-      rev = value.src.rev;
-    }} else null;
-  checksums = lib.mapAttrs getChecksum generated;
-in lib.filterAttrs (n: v: v != null) checksums)"""
+in
+lib.mapAttrs parse generated"""
 
 HEADER = "# This file has been generated by ./pkgs/applications/editors/kakoune/plugins/update.py. Do not edit!"
 
 class KakouneEditor(pluginupdate.Editor):
-
-
     def generate_nix(self, plugins: List[Tuple[pluginupdate.PluginDesc, pluginupdate.Plugin]], outfile: str):
-        sorted_plugins = sorted(plugins, key=lambda v: v[1].name.lower())
-
```

Comment:
I decided to move sorting to base class, because why duplicate logic? Furthermore, Vim and Kakoune sorted by different keys (also, Vim's sorting was broken)

Reply:
agreed